<?xml version="1.0"?>
<launch>
    <arg name="interface" default="can0"/>
    <arg name="frame_id" default="laser"/>
    <arg name="fov" default="100.0"/>
    <arg name="min_range" default="0.0"/>
    <arg name="max_range" default="50.0"/>
    <arg name="rate" default="50.0"/>
    <arg name="base_rx_message_id" default="1856"/>
    <arg name="base_tx_message_id" default="1872"/>
    <arg name="scan" default="/leddar_vu8/scan"/>
    <arg name="send_timeout" default="10.0"/>
    <arg name="receive_timeout" default="10.0"/>
    <arg name="respawn" default="false"/>
    <arg name="view" default="true"/>
    <arg name="name" default="leddar_vu8"/>

    <remap from="/$(arg name)/scan" to="$(arg scan)"/>

    <node name="$(arg name)"
          pkg="leddar_vu8" 
          type="leddar_vu8"
          output="screen"
          required="true"
          respawn="$(arg respawn)">
        <param name="interface" type="string" value="$(arg interface)"/>
        <param name="frame_id" type="string" value="$(arg frame_id)"/>
        <param name="fov" type="double" value="$(arg fov)"/>
        <param name="rate" type="double" value="$(arg rate)"/>
        <param name="min_range" type="double" value="$(arg min_range)"/>
        <param name="max_range" type="double" value="$(arg max_range)"/>
        <param name="base_rx_message_id" type="int" value="$(arg base_rx_message_id)"/>
        <param name="base_tx_message_id" type="int" value="$(arg base_tx_message_id)"/>
        <param name="send_timeout" type="double" value="$(arg send_timeout)"/>
        <param name="receive_timeout" type="double" value="$(arg receive_timeout)"/>
    </node>

    <node name="$(arg name)_viz"
          pkg="rviz" 
          type="rviz"
          args="-d $(find leddar_vu8)/launch/leddar_vu8.rviz"
          required="true"
          if="$(arg view)" />
</launch>
